edb/pgsql/codegen.py

```diff
@@ -111,6 +112,13 @@ def translate(self, pos: int) -> int:
         return self.source_start
 
 
+@dataclasses.dataclass(frozen=True)
+class SQLSource:
+    chunks: list[str]
+    param_index: dict[int, list[int]]
+    translation_data: Optional[SQLSourceGeneratorTranslationData] = None
```

Comment:
I've opened a PR that does a refactor of codegen, which will conflict with these changes.

The changes are in the same light (adding dataclasses as such), and should not be hard to update the these changes.